velox/exec/VectorHasher.h

```diff
@@ -172,6 +172,11 @@ class VectorHasher {
   // computeValueIds(). The decoded vector can be accessed via decodedVector()
   // getter.
   void decode(const BaseVector& vector, const SelectivityVector& rows) {
+    VELOX_CHECK(
+        type_->kindEquals(vector.type()),
+        "Type mismatch: {} vs. {}",
+        type_->toString(),
+        vector.type()->childAt(channel_)->toString());
     decoded_.decode(vector, rows);
```

Comment:
since calling decode() is a pre-req for calling hash() and other methods, should we add a check/enforcement for that too?

Reply:
That would be nice. Not sure how to do that though.